### ISSUE TITLE
Insecure $ENV{BASH_ENV} while running with -T switch

### ISSUE THREAD

**Issue by lapseofreason** [digint/btrbk#187]
Since about a week I get the error below on an installation that used to be working fine before. I think this appeared with the upgrade to Fedora 26 (Perl  v5.24.2). I can reproduce it both with git master and the version in Fedora (v0.25.1).

~~~~
ERROR: process died unexpectedly (btrbk v0.25.1)
Please contact the author: Axel Burri <axel@tty0.ch>

Stack Trace:
----------------------------------------
Insecure $ENV{BASH_ENV} while running with -T switch at /usr/sbin/btrbk line 710.
 at /usr/sbin/btrbk line 262.
	main::__ANON__("Insecure \$ENV{BASH_ENV} while running with -T switch at /usr/"...) called at /usr/sbin/btrbk line 710
	main::run_cmd("cmd", ARRAY(0xXXXXXXX), "rsh", undef, "non_destructive", 1, "catch_stderr", 1, ...) called at /usr/sbin/btrbk line 854
	main::btrfs_subvolume_show(HASH(0xXXXXXXX)) called at /usr/sbin/btrbk line 1965
	main::vinfo_init_root(HASH(0xXXXXXXX)) called at /usr/sbin/btrbk line 4523
~~~~

Removing the -T switch fixes the problem, but not being familiar with Perl I don't know what the security implications of this are. I'm happy to help if you need any additional information.

**Comment by digint** [comment]
Perl taint mode (-T switch) is a bitch. If enabled, perl refuses to execute anything of which has not explicitely been checked (that's why btrbk hardcodes $PATH for example). Now btrbk starts shell commands (e.g. "btrfs subvolume show"), and having BASH_ENV set means that the shell (bash in your setup) will also run the commands in `$BASH_ENV`, and perl cannot know if these commands are sane.

Note that disabling taint-mode is subject to change in future btrbk versions, this has already caused too much headache; see also: #178 

As a workaround, you could run btrbk with
```
# unset BASH_ENV
# btrbk run
```

Or you could add the following to btrbk (line 3704): 
```diff
   # different distros (and even different versions of btrfs-progs)
   # install the "btrfs" executable to different locations.
   $ENV{PATH} = '/sbin:/bin:/usr/sbin:/usr/bin';
+  delete $ENV{BASH_ENV};
  
   Getopt::Long::Configure qw(gnu_getopt);
   my $start_time = time;
```

Or simply disable "-T" switch to skip perl doing all the extraneous checks. If your environment is sane, there are no security implications.

**Comment by digint** [comment]
fixed in: 571dae4428d12b1ebba56a36b9b43fca28449781

Taint mode is now disabled by default.
